lesson11/index.js

```diff
@@ -0,0 +1,62 @@
+const ul = document.getElementById('js-lists');
+
+const fetchData = async() => {
+    try {
+        const response = await fetch('https://jsondata.okiba.me/v1/json/KN8s0211027113740');
+        const obj = await response.json();
+        const listData = obj.data;
+        return listData;
+    } catch (error) {
+        errorText(`エラーが発生しました: 詳細 ${error}`);
+    }
+}
+
+const displayLoadingImage = () => {
+    const loadingImage = document.createElement('img');
+    loadingImage.id = 'loadingImageId';
+    loadingImage.src = './loading-circle.gif';
+    ul.appendChild(loadingImage);
+}
+
+const removeLoadingImage = () => {
+    const loadingImage = document.getElementById('loadingImageId');
+    loadingImage.remove();
+}
+
+const displayContents = async() => {
+    displayLoadingImage();
+    try {
+        const data = await fetchData();
+        createList(data);
```

Comment:
nis
Think about what would happen if the data could not be fetched for some reason. This will make the code more friendly.
For example
If the fetched json data is
```json
{}
```
What if this was all there was?

```suggestion
           const data = await fetchData();
    if (data) {
      // if data is ture.
      // (null, undefined, 0, empty character (''), equivalent to if not false)
      createList(data);
    } else {
       // if data is null, undefined, 0, empty character (''), false
      errorText("エラーが発生しました: 詳細 dataがありません");
    }
```
If you write this, you'll notice that the acquisition went well, but the data didn't.

I wrote it in Japanese.
[codesandbox](https://codesandbox.io/s/serene-hodgkin-z2hp3?file=/index.js)

Reply:
I considered it in the previous issue, but I didn't consider it this time.
It seems difficult to think about what kind of error is expected.
Thank you so much!!